arcgishub/hub.py

```diff
@@ -1,182 +1,133 @@
+class Initiative(collections.OrderedDict):
+    """Represents an initiative"""
     
-    def indicator_delete(self, initiative_id, indicator_id):
-        '''Deletes a particular indicator for given initiative'''
-        _invalid_indicator = False
-        initiative = self.initiative_get(initiative_id)
-        data = initiative.get_data()
-        for indicator in data['indicators']:
-            if indicator_id==indicator['id']:
-                _invalid_indicator = True
-                break
-        if _invalid_indicator:
-            data['indicators'] = list(filter(lambda indicator: indicator.get('id')!=indicator_id, data['indicators']))
-            initiative_data = json.dumps(data)
-            return initiative.update(item_properties={'text': initiative_data})
+    def __init__(self, org, initiativeItem=None):
+        '''Constructs an empty Initiative object'''
+        if initiativeItem:
+            if 'hubInitiative' not in initiativeItem.typeKeywords:
+                raise TypeError("Item is not a valid initiative.")
+            self.item = initiativeItem
+            self._org = org
+            self._initiativedict = self.item.get_data()
+            pmap = PropertyMap(self._initiativedict)
+            self.definition = pmap
         else:
-            return 'Indicator does not exist'
-        
-    def indicator_get(self, initiative_id, indicator_id):
-        '''Fetch an indicator based on id'''
-        indicators = self._all_indicators(initiative_id)
-        try:
-            for indicator in indicators:
-                if indicator['id']==indicator_id:
-                    return indicator
-        except:
-            return 'Indicator does not exist'
-        
-    def indicator_search(self, initiative_id, indicator_id=None, url=None, itemId=None, name=None):
-        '''Search for indicator'''
-        indicators = []
-        temp = self._all_indicators(initiative_id)
-        for indicator in temp:
-            indicators.append(self._build_indicator_object(indicator))
-        if indicator_id!=None:
-            indicators = [indicator for indicator in indicators if indicator['id']==indicator_id]
-        if url!=None:
-            indicators = [indicator for indicator in indicators if indicator['url']==url]
-        if itemId!=None:
-            indicators = [indicator for indicator in indicators if indicator['itemId']==itemId]
-        if name!=None:
-            indicators = [indicator for indicator in indicators if indicator['name']==name]
-        return indicators
+            self.item = None
+            self._org = org
+            self._initiativedict = {"type":"Hub Initiative", "typekeywords":"hubInitiative"}
+            pmap = PropertyMap(self._initiativedict)
+            self.definition = pmap
+            
+    def __repr__(self):
+        return '<%s title:"%s" owner:%s>' % (type(self).__name__, self.initiativeTitle, self.initiativeOwner)
+       
+    @property
+    def initiativeId(self):
```

Comment:
is it redundant to say:

```python
initiative = Hub.initiatives.get('4ef')
initiative.initiativeId
```

this could be simpler to have `initiative.id` if that's valid Python

Reply:
`id` is a Python keyword so can't use it.
Another option is calling `initiative.item.id` or `initiative.item.owner` etc, but I'm not sure if that is better than `initiative.initiativeId`